### PR TITLE
feat(helm): add --reuse-values flag to upgrade

### PR DESCRIPTION
**Pull request by technosophos**
This makes it possible to re-use the existing values on an upgrade,
merging in any new values set by `-f` or `--set`.

Closes #1876 

**THIS NEEDS MANUAL TESTING**

This feature makes it possible to _keep your old values_, but _selectively override them_.

Say you installed with `helm install -f foo.yaml somechart`, where foo.yaml had hundreds of parameters. Now you want to change just one parameter without finding the old `foo.yaml` file.

To do this, you can:

```
$ helm upgrade --reuse-values --set name=value old_release foo
```

This should _start_ with the old values, and then merge the `--set` values into them.

So `helm get old_release` should show all of the values originally set, PLUS the `name: value` parameter.

### CODE REVIEW

File: cmd/helm/upgrade.go [hunk 1]
```diff
@@ -114,6 +115,7 @@ func newUpgradeCmd(client helm.Interface, out io.Writer) *cobra.Command {
 	f.StringVar(&upgrade.version, "version", "", "specify the exact chart version to use. If this is not specified, the latest version is used")
 	f.Int64Var(&upgrade.timeout, "timeout", 300, "time in seconds to wait for any individual kubernetes operation (like Jobs for hooks)")
 	f.BoolVar(&upgrade.resetValues, "reset-values", false, "when upgrading, reset the values to the ones built into the chart")
+	f.BoolVar(&upgrade.reuseValues, "reuse-values", false, "when upgrading, reuse the last release's values, and merge in any new values")
```
Comment (thomastaylor312) [inline cmd/helm/upgrade.go]: You may want to add the note in the help text here that this is ignored if `--reset-values` is set

**Review by thomastaylor312 — APPROVED** [review]
Tested manually and everything looks good.

File: cmd/helm/upgrade.go (hunk 1 above)
Comment (thomastaylor312) [inline cmd/helm/upgrade.go]: I am not going to block on this, but if you add this, no need to re-review

Comment (technosophos) [inline cmd/helm/upgrade.go]: fixed.

**Pull request merged**